File: app/src/main/java/com/example/natasaandzic/courtcounter/MainActivity.java
package com.example.natasaandzic.courtcounter;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    int scoreA=0;
    int scoreB=0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void displayScoreForA(int number) {
        TextView scoreA = findViewById(R.id.scoreA);
        scoreA.setText("" +number);
    }

    public void displayScoreForB(int number) {
        TextView scoreB = findViewById(R.id.scoreB);
        scoreB.setText("" +number);
    }

    public void plusThreeA(View view) {
        scoreA+=3;
        displayScoreForA(scoreA);
    }

    public void plusTwoA(View view) {
        scoreA+=2;
        displayScoreForA(scoreA);
    }

    public void plusOneA(View view) {
        scoreA+=1;
        displayScoreForA(scoreA);
    }

    public void plusThreeB(View view) {
        scoreB+=3;
        displayScoreForB(scoreB);
    }

    public void plusTwoB(View view) {
        scoreB+=2;
        displayScoreForB(scoreB);
    }

    public void plusOneB(View view) {
        scoreB+=1;
        displayScoreForB(scoreB);
    }

    public void reset(View view) {
        scoreA=0;
        scoreB=0;
        displayScoreForA(scoreA);
        displayScoreForB(scoreB);
    }
}

File: app/src/main/java/com/example/natasaandzic/courtcounter/MainActivity.kt
package com.example.natasaandzic.courtcounter

import android.os.Bundle

import android.support.v7.app.AppCompatActivity
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var scoreA = 0
    private var scoreB = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        aPlusThreeBtn.setOnClickListener {
            this.scoreA = increment(this.scoreA, 3)
            displayScore(scoreATv, this.scoreA)
        }

        aPlusTwoBtn.setOnClickListener {
            this.scoreA = increment(scoreA, 2)
            displayScore(scoreATv, this.scoreA)
        }

        aPlusOneBtn.setOnClickListener {
            this.scoreA = increment(scoreA, 1)
            displayScore(scoreATv, this.scoreA)
        }

        resetBtn.setOnClickListener {
            scoreA = 0
            scoreB = 0
            displayScore(scoreATv, 0)
            displayScore(scoreBTv, 0)
        }

        bPlusThreeBtn.setOnClickListener {
            this.scoreB = increment(scoreB, 3)
            displayScore(scoreBTv, this.scoreB)
        }

        bPlusTwoBtn.setOnClickListener {
            this.scoreB = increment(scoreB, 2)
            displayScore(scoreBTv, this.scoreB)
        }

        bPlusOneBtn.setOnClickListener {
           this.scoreB = increment(scoreB, 1)
            displayScore(scoreBTv, this.scoreB)
        }
    }

    private fun displayScore(view: TextView, score: Int) {
        view.text = score.toString()
    }

    private fun increment(number: Int, plus: Int): Int {
        return number + plus
    }

}